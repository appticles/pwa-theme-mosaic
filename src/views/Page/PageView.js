import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Container, Image, Header, Label, Button } from 'semantic-ui-react';
import Moment from 'react-moment';

import { pagePropType } from '../SideMenu/reducer';
import { closeMenu } from '../../components/NavBar/action';
import '../SideMenu/style.css';
import PageList from '../SideMenu/PageList';

class PageView extends Component {
  constructor(props) {
    super(props);
    this.hideSidebar = this.hideSidebar.bind(this);
  }

  hideSidebar() {
    if (this.props.sideMenuVisible) {
      this.props.closeMenu();
    }
  }

  render() {
    const { page } = this.props;
    // const { author } = page._embedded;
    // const categories = page._embedded['wp:term'];
    // const featuredMedia = page._embedded['wp:featuredmedia'];

    // function getImage(sourceImg) {
    //   let imageSource;
    //   if (sourceImg) {
    //     imageSource = sourceImg[0].source_url;
    //     return imageSource;
    //   }
    //   imageSource = 'https://placeholdit.co//i/555x650';
    //   return imageSource;
    // }
    console.log('props', this);
    return (
      <Container className="page">
        <Link to="/" onClick={this.hideSidebar}>
          BACK
          <Button circular icon="settings" />
        </Link>
        {/* <Image src={getImage(featuredMedia)} /> */}
        <Container textAlign="justified">
          {/* {categories[0].map(category => (
            <Label color="teal" key={Math.random()}>
              {category.name}
            </Label>
          ))} */}
          <Header>
            <div dangerouslySetInnerHTML={{ __html: page.title.rendered }} />
          </Header>
          {/*
            <Header.Subheader>
              &nbsp;by&nbsp;<b>{author[0].name}</b>,&nbsp;<Moment format="MMMM DD, YYYY">{page.date}</Moment>
            </Header.Subheader>
          */}
          <div dangerouslySetInnerHTML={{ __html: page.content.rendered }} />
        </Container>
      </Container>
    );
  }
}

PageView.propTypes = {
  page: pagePropType.isRequired,
  sideMenuVisible: PropTypes.bool.isRequired,
  closeMenu: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  sideMenuVisible: state.sideMenuVisible,
});
function mapDispatchToProps(dispatch) {
  return Object.assign(bindActionCreators({ closeMenu }, dispatch));
}
export default connect(mapStateToProps, mapDispatchToProps)(PageView);
