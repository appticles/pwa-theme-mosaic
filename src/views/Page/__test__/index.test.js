import React from 'react';
import renderer from 'react-test-renderer';

import mockStore from '../../../__mocks__/redux-mock-store';

import PageView from '../index';

jest.mock('../PageDetails', () => 'Page details test');
jest.mock('../../../components/NotFound/', () => 'Not found component test');
jest.mock('react-global-configuration', () => require.requireActual('../../../__mocks__/react-global-configuration').default);

describe('Page screen', () => {
  it('renders a message if the page does not exist', () => {
    const store = mockStore({
      pages: {
        items: [],
        isFetching: 0,
      },
      translations: {
        items: {},
        isFetching: 0,
      },
    });
    const match = {
      params: {
        pageId: '1',
      },
    };
    expect(renderer.create(<PageView store={store} fetchPages={jest.fn()} match={match} />)).toMatchSnapshot();
  });

  it('renders loader', () => {
    const store = mockStore({
      pages: {
        items: [],
        isFetching: 1,
      },
      translations: {
        items: {},
        isFetching: 0,
      },
    });
    const match = {
      params: {
        pageId: '1',
      },
    };
    expect(renderer.create(<PageView store={store} fetchPages={jest.fn()} match={match} />)).toMatchSnapshot();
  });

  it('renders existing pages', () => {
    const store = mockStore({
      pages: {
        items: [
          {
            id: 1,
            parent: 32,
            title: {
              rendered: 'Some page tile',
            },
            slug: 'some-page-slug',
            content: {
              rendered: 'Some page content ot be rendered',
            },
            link: 'some-page-link',
            author: 54,
            comment_status: 'dunno',
            featured_media: 43,
            status: 'some-status',
            _embedded: {
              author: [
                {
                  id: 54,
                  name: 'Some author',
                },
              ],
            },
          },
          {
            id: 3,
            parent: 34,
            title: {
              rendered: 'Some other page tile',
            },
            slug: 'some-other-page-slug',
            content: {
              rendered: 'Some page content ot be rendered',
            },
            link: 'some-page-link',
            author: 54,
            comment_status: 'dunno',
            featured_media: 43,
            status: 'some-status',
            _embedded: {
              author: [
                {
                  id: 54,
                  name: 'Some author',
                },
              ],
            },
          },
        ],
        isFetching: 0,
      },
      translations: {
        items: {
          TEXTS: {
            BY_AUTHOR: 'by',
            NO_PAGES: 'There are no pages!',
          },
        },
        isFetching: 0,
      },
    });
    const match = {
      params: {
        pageId: '1',
      },
    };
    expect(renderer.create(<PageView store={store} fetchPages={jest.fn()} match={match} />)).toMatchSnapshot();
  });
});
