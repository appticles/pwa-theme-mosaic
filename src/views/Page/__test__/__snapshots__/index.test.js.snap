// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page screen renders a message if the page does not exist 1`] = `
<div
  className="ui active loader"
/>
`;

exports[`Page screen renders existing pages 1`] = `
<Page details test
  page={
    Object {
      "author": 54,
      "comment_status": "dunno",
      "content": Object {
        "rendered": "Some page content ot be rendered",
      },
      "featured_media": 43,
      "id": 1,
      "link": "some-page-link",
      "parent": 32,
      "slug": "some-page-slug",
      "status": "some-status",
      "title": Object {
        "rendered": "Some page tile",
      },
    }
  }
  texts={
    Object {
      "TEXTS": Object {
        "BY_AUTHOR": "by",
        "NO_PAGES": "There are no pages!",
      },
    }
  }
/>
`;

exports[`Page screen renders loader 1`] = `
<div
  className="ui active loader"
/>
`;
